File: Data Structures & Algorithms/merge-strings-alternately/submission-0.py
class Solution:
    def mergeAlternately(self, s1: str, s2: str) -> str:
        s = ""

        while s1 and s2:
            s += s1[0] + s2[0]
            s1 = s1[1:]
            s2 = s2[1:]
        
        if s1:
            return s + s1
        else:
            return s + s2

File: Data Structures & Algorithms/last-stone-weight/submission-0.java
class Solution {
    public int lastStoneWeight(int[] stones) {
        PriorityQueue<Integer> pq = new PriorityQueue<>(Comparator.reverseOrder());
        for(int stone: stones) {
            pq.add(stone);
        }
        while(pq.size() > 1) {
            int first = pq.poll();
            int second = pq.poll();

            int diff = Math.abs(first - second);
            if(diff != 0) {
                pq.add(diff);
            }
        }

        return pq.isEmpty() ? 0 : pq.poll();
    }
}
